browser/src/core/Socket.js

```diff
@@ -1318,42 +1317,24 @@ app.definitions.Socket = L.Class.extend({
 	_delayMessage: function(textMsg) {
 		var message = {msg: textMsg};
 		this._delayedMessages.push(message);
-
-		if (!this._delayedMsgHandlerTimeoutId) {
-			this._handleDelayedMessages();
-		}
 	},
 
-	_handleDelayedMessages: function() {
-		if (!this._isReady() || !this._map._docLayer || this._handlingDelayedMessages) {
-			var that = this;
-			// Retry in a bit.
-			this._delayedMsgHandlerTimeoutId = setTimeout(function() {
-				that._handleDelayedMessages();
-			}, 100);
-			return;
-		}
+	_handleDelayedMessages: function(docLayer) {
 		var messages = [];
 		for (var i = 0; i < this._delayedMessages.length; ++i) {
 			var message = this._delayedMessages[i];
-			if (message)
-				messages.push(message.msg);
+			messages.push(message.msg);
 		}
 		this._delayedMessages = [];
```

Comment:
On that basis, I guess popping one message and processing it might be a better plan (?) - but then we would want the || this._delayedMessages.length on delay / queueing incoming messages (?)